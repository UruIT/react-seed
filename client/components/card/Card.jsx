import React from 'react';
import PropTypes from 'prop-types';
import styles from './card.scss';

const Card = props => (
	<div className={styles.card}>
		<h4>{props.title}</h4>
		<p>{props.content}</p>
	</div>
);

Card.propTypes = {
	content: PropTypes.string,
	title: PropTypes.string
};

Card.defaultProps = {
	content: 'Card content',
	title: 'Card title'
};

export default Card;
