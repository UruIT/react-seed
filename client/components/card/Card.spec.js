import React from 'react';
import renderer from 'react-test-renderer';


import Card from './Card';

describe('<Card />', () => {
	it('snapshot - default props', () => {
		const tree = renderer.create(<Card />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('snapshot - with content', () => {
		const tree = renderer.create(<Card content="TEXT" />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('snapshot - with title', () => {
		const tree = renderer.create(<Card title="TEXT" />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('snapshot - with title and content', () => {
		const tree = renderer.create(<Card content="CONTENT" title="TITLE"/>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
