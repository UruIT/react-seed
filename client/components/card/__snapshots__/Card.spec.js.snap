// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> snapshot - default props 1`] = `
<div
  className="card"
>
  <h4>
    Card title
  </h4>
  <p>
    Card content
  </p>
</div>
`;

exports[`<Card /> snapshot - with content 1`] = `
<div
  className="card"
>
  <h4>
    Card title
  </h4>
  <p>
    TEXT
  </p>
</div>
`;

exports[`<Card /> snapshot - with title 1`] = `
<div
  className="card"
>
  <h4>
    TEXT
  </h4>
  <p>
    Card content
  </p>
</div>
`;

exports[`<Card /> snapshot - with title and content 1`] = `
<div
  className="card"
>
  <h4>
    TITLE
  </h4>
  <p>
    CONTENT
  </p>
</div>
`;
